durabletask/worker.py

```diff
@@ -88,14 +89,36 @@ def __init__(self, *,
                  metadata: Optional[list[tuple[str, str]]] = None,
                  log_handler=None,
                  log_formatter: Optional[logging.Formatter] = None,
-                 secure_channel: bool = False):
+                 secure_channel: bool = False,
+                 access_token_manager: AccessTokenManager = None):
         self._registry = _Registry()
         self._host_address = host_address if host_address else shared.get_default_host_address()
         self._metadata = metadata
         self._logger = shared.get_logger("worker", log_handler, log_formatter)
         self._shutdown = Event()
         self._is_running = False
         self._secure_channel = secure_channel
+        self._access_token_manager = access_token_manager
+        self.__update_metadata_with_token()
+
+    def __update_metadata_with_token(self):
+        """
+        Add or update the `authorization` key in the metadata with the current access token.
+        """
+        if self._access_token_manager is not None:
+            token = self._access_token_manager.get_access_token()
+            
+            # Check if "authorization" already exists in the metadata
+            updated = False
+            for i, (key, _) in enumerate(self._metadata):
+                if key == "authorization":
+                    self._metadata[i] = ("authorization", token)
+                    updated = True
+                    break
+            
+            # If not updated, add a new entry
+            if not updated:
+                self._metadata.append(("authorization", token))
```

Comment:
The way we're building these SDKs is that the core functionality is general purpose and applicable to a variety of different backend implementations. For example, Dapr, DTS, or something else. However, this change here is specific to DTS (because it assumes an "authorization" header that requires a token) and therefore violates the design principle we're going for.  I would ask that you follow the example of the .NET SDK and create a separate package which is specific to DTS that is able to extend the core worker logic. It's okay if we also need to change `worker.py`, but only if it's a general purpose change that could potentially benefit multiple backends.

Reply:
@cgillum instead of `durabletask.client` and `durabletask.worker` would having `durabletask.dts.clients` and `durabletask.dts.worker` work? Those of course should utilize the former in their implementation - and then any Azure credential, token refresh and other logic can be added as well.

Reply:
Yeah, that's roughly what I was hoping we could do. So instead of a developer creating a `TaskHubGrpcWorker/Client` and passing in a bunch of generic parameters, they'd create a `DurableTaskSchedulerWorker/Client` defined in a new `durabletask-azuremanaged` package from PyPI, which has constructors that are specifically designed for use with DTS. Internally, they'd use the `durabletask.client` and `durabletask.worker` as you said.

Reply:
So you don't want it as a subpackage in the main `durabetask-python` library, but instead publish it as an entirely different package?

I guess that makes sense because we could avoid dependency bloat (for those who don't want to install Azure SDK dependencies like `azure-identity) when using durable task. That's the biggest advantage of the separate package.

I'll let you and @RyanLettieri decide - I can help review.

Reply:
Correct. This should be an entirely different package to avoid the dependency bloat. It's the same strategy we're using for the .NET and Java SDKs.